### PR TITLE
Refactor Copying Model Elements

### PR DESCRIPTION
**Pull request by philippfromme**
Requires https://github.com/bpmn-io/diagram-js/pull/390
Requires https://github.com/bpmn-io/bpmn-js/pull/1137

Closes https://github.com/bpmn-io/bpmn-js/issues/1138
Closes https://github.com/bpmn-io/bpmn-js/issues/1139

### CODE REVIEW

File: test/spec/util/CopyModelElementUtilSpec.js [hunk 1]
```diff
@@ -507,6 +547,26 @@ describe('util/CopyModelElementUtil', function() {
...
+        if (is(sourceElement, 'bpmn:Task')) {
+          return false;
+        }
+      });
+
+      var userTask = copyModelElementUtil.copyElement(task, moddle.create('bpmn:UserTask'));
+
+      expect(userTask.name).not.to.exist;
+    }));
+
```
Comment (nikku) [inline test/spec/util/CopyModelElementUtilSpec.js]: Add space.

File: lib/features/copy-paste/BpmnCopyPaste.js [hunk 2]
```diff
@@ -69,6 +72,18 @@ export default function BpmnCopyPaste(bpmnFactory, bpmnRules, eventBus, injector
     }
   });
 
+  eventBus.on('element.copyProperty', function(context) {
+    var parent = context.parent,
+        property = context.property,
+        propertyName = context.propertyName;
+
+    if (is(parent, 'bpmn:Participant') &&
```
Comment (nikku) [inline lib/features/copy-paste/BpmnCopyPaste.js]: Can we reformat this for better visibility:

```
if (
  abcde &&
  efasdads &&
  asdsaasdadsdas
) {
  ...
}
```

Comment (philippfromme) [reply lib/features/copy-paste/BpmnCopyPaste.js]: 👍

**Comment by philippfromme** [comment]
> Do we have test cases that verify the copied `bpmn:Group#categoryValueRef` is wired to the model on paste?

I have to check.

File: test/spec/util/CopyModelElementUtilSpec.js (hunk 1 above)
Comment (philippfromme) [reply test/spec/util/CopyModelElementUtilSpec.js]: 🔍

**Comment by philippfromme** [comment]
> > Do we have test cases that verify the copied `bpmn:Group#categoryValueRef` is wired to the model on paste?
> 
> I have to check.

Tests ✔️

**Comment by barmac** [comment]
`package-lock.json` wasn't updated in https://github.com/bpmn-io/bpmn-js/pull/1141/commits/49ff06f39d256b55ac959742a8c4134dc1b9ee7c which causes `npm ci` to fail.

**Comment by barmac** [comment]
Got
```
Cannot read property 'length' of undefined
    at removeReferencedCategoryValue (webpack-internal:///../../bpmn-js/lib/features/modeling/behavior/GroupBehavior.js:53:45)
    at eval (webpack-internal:///../../bpmn-js/lib/features/modeling/behavior/GroupBehavior.js:110:9)
    at invokeFunction (webpack-internal:///../../diagram-js/lib/core/EventBus.js:472:13)
    at EventBus._invokeListener (webpack-internal:///../../diagram-js/lib/core/EventBus.js:332:19)
    at EventBus._invokeListeners (webpack-internal:///../../diagram-js/lib/core/EventBus.js:320:24)
    at EventBus.fire (webpack-internal:///../../diagram-js/lib/core/EventBus.js:284:24)
    at CommandStack._fire (webpack-internal:///../../diagram-js/lib/command/CommandStack.js:339:29)
    at eval (webpack-internal:///../../diagram-js/lib/command/CommandStack.js:399:10)
    at CommandStack._atomicDo (webpack-internal:///../../diagram-js/lib/command/CommandStack.js:358:5)
    at CommandStack._internalExecute (webpack-internal:///../../diagram-js/lib/command/CommandStack.js:388:8)
```
when trying to remove all of the groups at once. Not sure if this is related to copy paste.

**Comment by philippfromme** [comment]
Closed in favor of https://github.com/bpmn-io/bpmn-js/pull/1137.

**Pull request closed without merging**